pkg/k8shandler/kibana/secret.go

```diff
@@ -27,35 +27,45 @@ var secretCertificates = map[string]map[string]string{
 func (clusterRequest *KibanaRequest) GetSecret(secretName string) (*core.Secret, error) {
 	secret := &core.Secret{}
 	if err := clusterRequest.Get(secretName, secret); err != nil {
-		if errors.IsNotFound(err) {
+		if apierrors.IsNotFound(kverrors.Root(err)) {
 			return nil, err
 		}
-		return nil, fmt.Errorf("Failed to get %v secret: %v", secret.Name, err)
+		return nil, kverrors.Wrap(err, "failed to get secret",
+			"name", secret.Name,
+		)
 	}
 
 	return secret, nil
 }
 
-func (clusterRequest *KibanaRequest) readSecrets() (err error) {
+// readSecrets reads all of the secrets it can find within secretCertificates
+// if any of the secrets are not found then they are ignored
+func (clusterRequest *KibanaRequest) readSecrets() error {
 
 	for secretName, certMap := range secretCertificates {
-		if err = clusterRequest.extractCertificates(secretName, certMap); err != nil {
-			return
+		err := clusterRequest.extractCertificates(secretName, certMap)
+		if err == nil {
+			continue
+		}
+		if !apierrors.IsNotFound(kverrors.Root(err)) {
+			return err
 		}
 	}
 
 	return nil
 }
 
-func (clusterRequest *KibanaRequest) extractCertificates(secretName string, certs map[string]string) (err error) {
+func (clusterRequest *KibanaRequest) extractCertificates(secretName string, certs map[string]string) error {
 
 	for secretKey, certPath := range certs {
-		if err = clusterRequest.extractSecretToFile(secretName, secretKey, certPath); err != nil {
-			if errors.IsNotFound(err) {
-				return nil
-			}
-			return
+		err := clusterRequest.extractSecretToFile(secretName, secretKey, certPath)
+		if err == nil {
+			return nil
 		}
+		return kverrors.Wrap(err, "failed to extract cert",
+			"key", secretKey,
+			"cert_path", certPath,
+		)
```

Comment:
isn't this different logic from we're using before... we now return an error if it isn't found